File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        unordered_map<string, vector<string>> mpp;

        for(auto str: strs){
            string hashStr="";
            int freq[26]={0};
            for(char s : str) freq[s-'a']++;
            for(int i=0;i<26;i++){
                hashStr.push_back(freq[i]);
                hashStr.push_back('#');
            }
            mpp[hashStr].push_back(str); 
        }

        vector<vector<string>> anagram;
        for(auto it:mpp){
            anagram.push_back(it.second);
        }
        return anagram;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """


        result=defaultdict(list)

        for str in strs:
            count=[0]*26

            for c in str:
                count[ord(c)-ord('a')]+=1
            
            result[tuple(count)].append(str)

        return list(result.values())

        